### PR TITLE
chore(compass-connections): Move connection modal UI from compass-sidebar to compass-connections COMPASS-8471

### PR DESCRIPTION
**Pull request by lerouxb**

### CODE REVIEW

File: packages/compass-connections/src/stores/connections-store-redux.ts [hunk 1]
```diff
@@ -2178,3 +2178,10 @@ export function configureStore(
     applyMiddleware(thunk.withExtraArgument(thunkArg))
   );
 }
+
+export type ConnectionsStore = ReturnType<typeof configureStore> extends Store<
```
Comment (lerouxb) [inline packages/compass-connections/src/stores/connections-store-redux.ts:2182]: will probably move this back  - don't think I needed to export it in the end

File: packages/compass-connections/src/plugin.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    </ConnectionActionsProvider>
+  );
+};
+
+function mapState({
+  editingConnectionInfoId,
+}: {
+  editingConnectionInfoId?: ConnectionId;
+}) {
+  return { editingConnectionInfoId };
```
Comment (lerouxb) [inline packages/compass-connections/src/plugin.tsx:47]: I'll probably undo adding this whole file and just do the check inside the modal.

File: packages/compass/src/app/components/home.tsx [hunk 3]
```diff
@@ -8,22 +8,15 @@ import {
   getScrollbarStyles,
   palette,
   resetGlobalCSS,
-  useEffectOnChange,
 } from '@mongodb-js/compass-components';
-import CompassConnections, {
-  SingleConnectionForm,
```
Comment (lerouxb) [inline packages/compass/src/app/components/home.tsx:14]: I should probably remove the single connection code in a separate PR. Was just temporarily doing it here to be able to see more easily what's going on.

File: packages/compass/src/app/components/home.tsx [hunk 4]
```diff
@@ -173,60 +134,37 @@ function Home({
...
-  );
-
   return (
-    <FileInputBackendProvider createFileInputBackend={createFileInputBackend}>
-      <ConnectionImportExportProvider>
-        <CompassInstanceStorePlugin>
-          <FieldStorePlugin>
-            <div data-testid="home" className={verticalSplitStyles}>
-              {multiConnectionsEnabled && (
+    <CompassConnections
```
Comment (lerouxb) [inline packages/compass/src/app/components/home.tsx:138]: Is this right now? The modal still doesn't seem to appear in the right place in the DOM.

File: packages/compass-welcome/src/components/welcome-tab.tsx [hunk 5]
```diff
@@ -218,7 +218,7 @@ function WelcomeImage() {
 }
 
 export default function WelcomeTab() {
-  const { createNewConnection } = useConnections();
+  const { createNewConnection } = useConnectionActions();
```
Comment (lerouxb) [inline packages/compass-welcome/src/components/welcome-tab.tsx:221]: This one was a drive-by. I don't think there's any reason to use the deprecated useConnections() here.

File: packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx [hunk 6]
```diff
@@ -107,28 +104,18 @@ export function MultipleConnectionSidebar({
...
-    saveAndConnect,
     disconnect,
     createNewConnection,
     editConnection,
-    cancelEditConnection,
     removeConnection,
     duplicateConnection,
-    saveEditedConnection,
-    toggleConnectionFavoritedStatus,
+    toggleFavoritedConnectionStatus,
```
Comment (lerouxb) [inline packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx:116]: was toggleConnectionFavoritedStatus in useConnections

File: packages/compass/src/app/components/home.tsx [hunk 7]
```diff
@@ -286,7 +312,7 @@ export function ThemedHome(
                 )}
                 data-theme={darkMode ? 'Dark' : 'Light'}
               >
-                <Home {...props}></Home>
+                <HomeWithConnections {...props}></HomeWithConnections>
```
Comment (lerouxb) [inline packages/compass/src/app/components/home.tsx:315]: Hierarchy is now ThemedHome -> HomeWithConnections -> Home whereas it was HomeWithConnections -> ThemedHome -> Home.

This allows CompassConnections to be inside all the visual stuff but still wrap the rest of home and therefore the useSingleConnectionModeConnectionInfoStatus() hook still works for now.

File: packages/compass-connections/src/components/connection-modal.tsx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+const mapDispatch = {
+  connect,
+  disconnect,
+  cancelEditConnection,
+  saveEditedConnectionInfo,
+  saveAndConnect,
+  openSettingsModal,
+};
+
+// TODO: surely there's a way of inferring these types?
```
Comment (lerouxb) [inline packages/compass-connections/src/components/connection-modal.tsx:66]: Any ideas?

**Review by gribnoysup — APPROVED** [review]
